multi-monitor support

adamhotep opened the pull request:
Added (slightly better) support for multiple monitors. When `sxiv` goes full-screen, it is on a per-_monitor_ basis. Adding `-Z` ensures that if you're on the top-left monitor, this works as expected. Monitors will open full-screen views of the screenshot with the top left pixel of the _monitor_ showing the top left pixel of the screenshot, so you'll see the wrong content on any other monitor.

I also used `trap` to ensure `$tmpfile` is cleaned up regardless of whether the script is terminated. 

Nice observations on speed! `maim` is a _lot_ faster than ImageMagick's `import` and `sxiv` is quite speedy as well. The script will be slightly faster using `/bin/sh` over `/bin/bash` (which would be slightly faster than `/usr/bin/env bash`), and since there are no bashisms, the more basic POSIX shell is functionally identical.

adamhotep committed:
multi-monitor support

Added support for multiple monitors. While sxiv defaults to full-screen, this is limited to the current monitor. Adding `-Z` ensures that if you're on the top-left monitor, this works as expected. Monitors will open full-screen views of the screenshot with the top left pixel of the _monitor_ showing the top left pixel of the screenshot, which is only appropriate on the top-left-most monitor.

I also used `trap` to ensure `$tmpfile` is cleaned up regardless of whether the script is terminated. 

Nice observations on speed! `maim` is a _lot_ faster than ImageMagick's `import` and `sxiv` is quite speedy as well. The script will be slightly faster using `/bin/sh` over `/bin/bash` (which would be slightly faster than `/usr/bin/env bash`), and since there are no bashisms, the more basic POSIX shell is functionally identical.

yassinebridi commented:
These are some really good remarks, things that I haven't thought about.

One thing though, have you checked if this script is working without any error?
Apparently the trap command fails to execute the removing of the file, and just throws an error instantly.

adamhotep committed:
removed CHLD signal trigger for the trap

adamhotep commented:
Yeah, you caught me having not tested my code. I'm not sure how the CHLD signal got in there, it triggered on each child process termination (including itself‽), which was not desired. I've fixed that and actually tested. I also added `-f` just to be sure it doesn't generate an error, though that shouldn't happen any more.

yassinebridi commented:
Thank you @adamhotep for this change.
